src/NServiceBus.Persistence.DynamoDB/SynchronizedStorage/ILockCleanup.cs

```diff
@@ -0,0 +1,16 @@
+namespace NServiceBus.Persistence.DynamoDB
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Amazon.DynamoDBv2;
+
+    interface ILockCleanup
+    {
+        Guid Id { get; }
+
+        bool NoLongerNecessaryWhenSessionCommitted { get; set; }
+
+        Task Cleanup(IAmazonDynamoDB client, CancellationToken cancellationToken = default);
```

Comment:
random thought: while against the guidance, I wonder whether we really need a cancellation token given it seems to be a type of action that should not be cancelable

Reply:
We discussed this and concluded from an implementation perspective the cleanup actions can handle cancellation and it is a top level concern to decide whether an action is cancellable. Floating the token through seems to make sense.